File: BACK_END_SPRING/src/main/java/ISM/entities/Etudiant.java
package ISM.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Table(name = "etudiants")
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter

@Entity
public class Etudiant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nom;

    @Column(nullable = false)
    private String prenom;
    @Column(nullable = false)
    private String tuteur;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String password;

    @Column(nullable = false)

    private String libelleClasse;

}

File: lib/models/etudiant.dart
class Etudiant {
  final String nom;
  final String prenom;
  final String tuteur;
  final String email;
  final String password;
  final String libelleClasse;

  Etudiant({
    required this.nom,
    required this.prenom,
    required this.tuteur,
    required this.email,
    required this.password,
    required this.libelleClasse,
  });

  factory Etudiant.fromJson(Map<String, dynamic> json) {
    return Etudiant(
      nom: json['nom'],
      prenom: json['prenom'],
      tuteur: json['tuteur'],
      email: json['email'],
      password: json['password'],
      libelleClasse: json['libelleClasse'],
    );
  }
}
